src/scaffoldmaker/meshtypes/meshtype_3d_wholebody1.py

```diff
@@ -0,0 +1,435 @@
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.cylindermesh import CylinderMesh, CylinderShape, CylinderEnds, CylinderCentralPath
+from scaffoldmaker.utils.zinc_utils import exnodeStringFromNodeValues
+from scaffoldmaker.scaffoldpackage import ScaffoldPackage
+from scaffoldmaker.meshtypes.meshtype_1d_path1 import MeshType_1d_path1
+from opencmiss.zinc.node import Node
+from opencmiss.zinc.element import Element
+from scaffoldmaker.annotation.annotationgroup import AnnotationGroup, findOrCreateAnnotationGroupForTerm, getAnnotationGroupForTerm, mergeAnnotationGroups
+from scaffoldmaker.annotation.body_terms import get_body_term
+from opencmiss.utils.zinc.general import ChangeManager
+from opencmiss.zinc.field import Field
+from opencmiss.utils.zinc.finiteelement import get_element_node_identifiers
+from scaffoldmaker.utils.eft_utils import remapEftNodeValueLabelsVersion
+
+
+class MeshType_3d_wholebody1(Scaffold_base):
+    """
+Generates body coordinates using a solid cylinder of all cube elements,
+ with variable numbers of elements in major, minor, shell and for each section of abdomen, thorax, neck and head.
+    """
+    cylinder1Settings = {
+                'Coordinate dimensions': 3,
+                'D2 derivatives': True,
+                'D3 derivatives': True,
+                'Length': 3.5,
+                'Number of elements': 1
+            }
+
+    centralPathDefaultScaffoldPackages = {
+        'Cylinder 1': ScaffoldPackage(MeshType_1d_path1, {
+            'scaffoldSettings': cylinder1Settings,
+            'meshEdits': exnodeStringFromNodeValues(
+                [Node.VALUE_LABEL_VALUE, Node.VALUE_LABEL_D_DS1, Node.VALUE_LABEL_D_DS2, Node.VALUE_LABEL_D2_DS1DS2,
+                 Node.VALUE_LABEL_D_DS3, Node.VALUE_LABEL_D2_DS1DS3], [
+                    [[0.0, 0.0, 0.0], [cylinder1Settings['Length'], 0.0, 0.0], [0.0, 0.5, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.5], [0.0, 0.0, 0.0]],
+                    [[cylinder1Settings['Length'], 0.0, 0.0], [cylinder1Settings['Length'], 0.0, 0.0], [0.0, 0.5, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, 0.5], [0.0, 0.0, 0.0]]
+                ])
+        })
+    }
+
+    @staticmethod
+    def getName():
+        return '3D Whole Body 1'
+
+    @classmethod
+    def getDefaultOptions(cls, parameterSetName='Default'):
+        centralPathOption = cls.centralPathDefaultScaffoldPackages['Cylinder 1']
+        options = {
+            'Central path': copy.deepcopy(centralPathOption),
+            'Number of elements across major': 6,
+            'Number of elements across minor': 6,
+            'Number of elements across shell': 1,
+            'Number of elements across transition': 1,
+            'Number of elements in abdomen': 5,
+            'Number of elements in thorax': 3,
+            'Number of elements in neck': 1,
+            'Number of elements in head': 2,
+            'Shell element thickness proportion': 1.0,
```

Comment:
'Shell element thickness proportion' is an option for the solid cylinder which I'm happy with. Here what you want is a 'Shell thickness proportion' so we can guarantee the proportion in the shell irrespective of the number of elements across it, and thus we control the body coordinates give to the core boundary. Take the parameter set here and back calculate the element thickness proportion to pass to the cylinder code.

Reply:
The cylinder mesh only allows the same 'Shell element thickness proportion' in major and minor directions. We can either have the same 'shell element thickness proportion' or the same 'sell thickness proportion' in major and minor directions if the number of elements across major and minor is not the same. If these are the same then both can happen at the same time.